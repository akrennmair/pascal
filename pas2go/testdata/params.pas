program test;

var o : integer;
	p : real;
	q : string;
	r, s : boolean;
	t : char;

procedure x(a : integer; b : real; c : string; d, e : boolean; f : char);
begin
	a := 1;
	b := 23.5;
	c := 'hello world!';
	d := true;
	e := false;
	f := 'X'
end;

begin
	x(o, p, q, r, s, t)
end.
