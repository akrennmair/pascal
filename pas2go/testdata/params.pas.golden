package main

import (
	"github.com/akrennmair/pascal/pas2go/system"
)

var _ = system.Write

// program test
func main() {
	var (
		o int
		p float64
		q string
		r bool
		s bool
		t byte
	)
	_ = o
	_ = p
	_ = q
	_ = r
	_ = s
	_ = t

	var x func(a int, b float64, c string, d bool, e bool, f byte)
	x = func(a int, b float64, c string, d bool, e bool, f byte) {
		a = 1
		b = 23.5e0
		c = "hello world!"
		d = true
		e = false
		f = 'X'
		return
	}

	x(o, p, q, r, s, t)
}
